scaleway/resource_compute_instance_server.go

```diff
@@ -0,0 +1,297 @@
+				Description: "the state of the server should be (started, stopped, standby)",
+				ValidateFunc: validation.StringInSlice([]string{
+					"started",
+					"stopped",
+					"standby",
+				}, false),
+			},
+			"user_data": {
+				Type: schema.TypeMap,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Optional:    true,
+				Description: "The user data associated with the server", // TODO: document reserved keys (`cloud-init`)
+			},
+			"zone":       zoneSchema(),
+			"project_id": projectIDSchema(),
+		},
+	}
+}
+
+func resourceScalewayComputeInstanceServerCreate(d *schema.ResourceData, m interface{}) error {
+	instanceApi, zone, err := getInstanceAPIWithZone(d, m)
+	if err != nil {
+		return err
+	}
+
+	name, ok := d.GetOk("name")
+	if !ok {
+		name = getRandomName("srv")
+	}
+	req := &instance.CreateServerRequest{
+		Zone:           zone,
+		Name:           name.(string),
+		Organization:   d.Get("project_id").(string),
+		Image:          d.Get("image").(string),
+		CommercialType: d.Get("type").(string),
+		EnableIPv6:     d.Get("enable_ipv6").(bool),
+		SecurityGroup:  d.Get("security_group_id").(string),
+	}
+
+	if raw, ok := d.GetOk("tags"); ok {
+		for _, tag := range raw.([]interface{}) {
+			req.Tags = append(req.Tags, tag.(string))
+		}
+	}
+
+	//if vs, ok := d.GetOk("root_volume"); ok {
+	//	req.Volumes = make(map[string]*instance.VolumeTemplate)
+	//	req.Volumes["0"] = &instance.VolumeTemplate{
+	//		ID:   v.(string),
+	//		Name: namesgenerator.GetRandomName(),
+	//	}
+	//}
+
+	res, err := instanceApi.CreateServer(req)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(newZonedId(zone, res.Server.ID))
+
+	// todo: add userdata
+
+	// room for improvement: start the action in Create / Update, wait for the state in Read
+	err = instanceApi.ServerActionAndWait(&instance.ServerActionAndWaitRequest{
+		Zone:     zone,
+		ServerID: res.Server.ID,
+		Action:   stateToAction(d.Get("state").(string)),
+		Timeout:  time.Minute * 10,
+	})
+	if err != nil {
+		return err
+	}
+
+	return resourceScalewayComputeInstanceServerRead(d, m)
+}
+
+func resourceScalewayComputeInstanceServerRead(d *schema.ResourceData, m interface{}) error {
+	instanceApi, zone, ID, err := getInstanceAPIWithZoneAndID(m, d.Id())
+	if err != nil {
+		return err
+	}
+
+	response, err := instanceApi.GetServer(&instance.GetServerRequest{
+		Zone:     zone,
+		ServerID: ID,
+	})
+	if err != nil {
+		if is404Error(err) {
+			d.SetId("")
+			return nil
+		}
+		return err
+	}
+
+	d.Set("name", response.Server.Name)
+	d.Set("image", response.Server.Image.ID)
+	d.Set("type", response.Server.CommercialType)
+	d.Set("tags", response.Server.Tags)
+	d.Set("security_group_id", response.Server.SecurityGroup.ID)
+
+	d.Set("enable_ipv6", response.Server.EnableIPv6)
+	d.Set("private_ip", *response.Server.PrivateIP)
+	d.Set("public_ip", response.Server.PublicIP.Address.String())
+
+	if response.Server.EnableIPv6 && response.Server.IPv6 != nil {
```

Comment:
What If ipv6 was enable and we disable it we should still Set public_ipv6 to ""

Reply:
let's keep default false for now